### PR TITLE
STYLE: specify encodings when opening files #52963

### LINKED ISSUE #52963: STYLE: specify encodings when opening files
Opening a file without an encoding can cause issues on Windows, there's been some reports from people trying to contribute on Windows running into this

pylint has a check for this, so let's enable it (I've been learning some Rust, so I'll try to get it in to Ruff so it can run in a reasonable amount of time and we can later use that instead - but for now a manual pylint check is fine)

Task here is:

1. make the following change to `.pre-commit-config.yaml`

```diff
diff --git a/.pre-commit-config.yaml b/.pre-commit-config.yaml
index 43b3699907..34cd91940f 100644
--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -83,9 +83,6 @@ repos:
     hooks:
     -   id: pylint
         stages: [manual]
--   repo: https://github.com/pycqa/pylint
-    rev: v2.16.2
-    hooks:
     -   id: pylint
         alias: redefined-outer-name
         name: Redefining name from outer scope
@@ -99,6 +96,11 @@ repos:
             |^pandas/conftest\.py  # keep excluded
         args: [--disable=all, --enable=redefined-outer-name]
         stages: [manual]
+    -   id: pylint
+        alias: unspecified-encoding
+        name: Using open without explicitly specifying an encoding
+        args: [--disable=all, --enable=unspecified-encoding]
+        stages: [manual]
 -   repo: https://github.com/PyCQA/isort
     rev: 5.12.0
     hooks:
```

2. run `pre-commit run unspecified-encoding --hook-stage manual --all-files`
3. fixup all the violations it reports (by using `encoding='utf-8'`), stage, commit, push, open pull request, celebrate

No need to ask for permission to work on this:
- just comment `take` and it will be assigned to you
- if someone else has commented `take` a week has passed and they haven't done anything (a common occurrence, unfortunately), then feel free to take over. Please only comment `take` if you intend to work on this

### PR DESCRIPTION
**Pull request by KetuPatel806**
- [x] closes #52963 (Replace #52963 with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.

### CODE REVIEW

File: b/.pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,432 @@
+minimum_pre_commit_version: 2.15.0
```
Comment (MarcoGorelli) [inline b/.pre-commit-config.yaml:1]: what's this file? could you remove it please?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -99,6 +96,11 @@ repos:
             |^pandas/conftest\.py  # keep excluded
         args: [--disable=all, --enable=redefined-outer-name]
         stages: [manual]
+    -       id: pylint
+            alias: unspecified-encoding
+            name: Using open without explicitly specifying an encoding
+            args: [--disable=all, --enable=unspecified-encoding]
+            stages: [manual]
```
Comment (MarcoGorelli) [inline .pre-commit-config.yaml:103]: could you keep the alignment that there is in the rest of the file?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
thanks for working on this

does `pre-commit run unspecified-encoding --hook-stage manual` pass?

File: b/.pre-commit-config.yaml (hunk 1 above)
Comment (KetuPatel806) [reply b/.pre-commit-config.yaml:1]: if i remove it. it given some error instruction of that line.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (KetuPatel806) [reply .pre-commit-config.yaml:103]: i think alignment are perfect for rest of the file

Comment (KetuPatel806) [reply .pre-commit-config.yaml:103]: pre-commit run unspecified-encoding --hook-stage manual which was skipped

Comment (MarcoGorelli) [reply .pre-commit-config.yaml:103]:
```suggestion
    -   id: pylint
        alias: unspecified-encoding
        name: Using open without explicitly specifying an encoding
        args: [--disable=all, --enable=unspecified-encoding]
        stages: [manual]
```

File: b/.pre-commit-config.yaml (hunk 1 above)
Comment (MarcoGorelli) [reply b/.pre-commit-config.yaml:1]: please unstage this file

File: .pre-commit-config.yaml (hunk 2 above)
Comment (KetuPatel806) [reply .pre-commit-config.yaml:103]: i'll do it.

File: b/.pre-commit-config.yaml (hunk 1 above)
Comment (KetuPatel806) [reply b/.pre-commit-config.yaml:1]: sure

File: .pre-commit-config.yaml (hunk 2 above)
Comment (KetuPatel806) [reply .pre-commit-config.yaml:103]: is there are no need of b/.pre-commit-config.yaml this file??

File: b/.pre-commit-config.yaml (hunk 1 above)
Comment (KetuPatel806) [reply b/.pre-commit-config.yaml:1]: is there no need of b/.pre-commit-config.yaml this file??

**Pull request closed without merging**